src/ICeWall/doc/add_pad.md

```diff
@@ -0,0 +1,51 @@
+## add_pad
+
+### Synopsis
+```
+  % add_pad [-name <name>] \
+                    [-type <type>] \
+                    [-cell <library_cell>] \
+                    [-signal <signal_name>] \
+                    [-edge (bottom|right|top|left)] \
+                    [-location {(centre|origin) {x <value> y <value>} [orient (R0|R90|R180|R270|MX|MY|MXR90|MYR90)]}] \
+                    [-bump {row <number> col <number>}] \
```

Comment:
In the add_pad command, “bump” vs “–bump” does not seem to make a difference in the result; 
This should probably generate an error with options are present that should not be there. Ie. bump requires the -bump

Reply:
I think when you call the `sta::parse_key_args` it doesn't generate an error, but instead leaves the arguments behind in the `args` variable. One way to check this would be to add `sta::check_argc_eq0 "add_pad" $args` to `add_pad`, `define_pad_cell`, `set_padring_options`, etc after you call the `parse_key_args`. to ensure that args is empty and generate an error message if it is not.

For example:
```
add_pad -edge bottom \
  -signal DVDD \
  -type vddiobias \
  -location {centre {x 310 y 115.0}} \
  -bondpad {centre {x 310 y 115.0}} \
  -inst_name PAD_DVDD_0_SOUTH
```
vs. 
```
add_pad -edge bottom \
  -signal DVDD \
  -type vddiobias \
  -location {centre {x 310 y 115.0}} \
  bondpad {centre {x 310 y 115.0}} \
  -inst_name PAD_DVDD_0_SOUTH
```
both work, but in the second one the bondpad is not actually defined since it got ignored.

Does this clarify it?

Reply:
Peter is correct. You should always check the positional arg count after parsing keyword/flag args.

I thought this was getting renamed to place_pad.

Reply:
Thanks, I see the issue now. I've added checks for positional arguments - this brought to light that I had been specifying -physical_only 1, whereas -physical_only is a flag, so no value is associated with it - so needed to change where I had used that.

I dont want the command to be called place_pad, because
1. After calling the command, the ad is not placed, it only becomes placed after running initialize_padring
2. I want to be able to enhance the command to make -location into an optional argument, and have the code determine the location for these cells automatically - again, this would only happen during initialize_padring.

so place_pad would imply something that is not true about what the command does. This is not th case for place_cell, which does result in the cell being placed immediately after the command has finished.